### PR TITLE
Pipeline improvements

### PR DESCRIPTION
**Pull request by xileF1337**
This PR contains several improvements to the LazyB/MuCHSALSA pipeline. Changes include:

- prevent race condition during index creation of input files when running multiple instances
- use the recommended bloom filter mode of Abyss instead of the legacy mpi mode
- skip non-unique k-mers in Jellyfish, greatly reducing the memory requirements
- Conda environment and Vagrant files added to provide the dependencies for using / compiling MuCHSALSA
- many sanity checks added: argument count, existence of input files, availability of external tools, abort on errors, status messages
- allow setting thread count and memory size of Bloom filters from the command line
- use quotes for variables to support paths containing spaces
- require python3 in shebang lines
- extended .gitignore file

### CODE REVIEW

**Comment by xileF1337** [comment]
@TGatter, would you be so kind as to review and merge? Thank you so much!

**Comment by 0x002A** [comment]
Thanks for the PR.

But before @TGatter merges, I'd like to give it a short review too. 😀

**Comment by xileF1337** [comment]
Sure, you're very welcome. My reference to Thomas was mainly due to the fact that I discussed those things with him already.

**Comment by xileF1337** [comment]
And just for clarification, this PR only contains technical improvements to the pipeline scripts, and no improvements to the actual assembly process. Thomas has to decide what exactly we will do about that.

File: CMakeLists.txt
Comment (0x002A) [inline CMakeLists.txt:1]: Is there a specific reason to downgrade the cmake version requirement here?

**Comment by 0x002A** [comment]
Only a single remark/question otherwise it lgtm. Will approve after clarification.

Nice code quality, good job

File: CMakeLists.txt
Comment (xileF1337) [reply CMakeLists.txt:1]: The pragmatic reason was that I only had version 3.16 of cmake available in the base image (Ubuntu focal, providing [v3.16.3](https://packages.ubuntu.com/focal/cmake)) that I used to provision the virtual machine via Vagrant. I used a VM to do the build before you fixed the issues with more recent compiler and libc++ versions. 

Anyway, I did not notice any problems with the v3.16.0 requirement. Was there any specific reason for you to enforce v3.17.0, or was this just the version you had at hand?

**Comment by xileF1337** [comment]
> Only a single remark/question otherwise it lgtm. Will approve after clarification.
> 
> Nice code quality, good job

Thanks for the review, @0x002A ! I commented on the Cmake version requirement in your remark above. Looking forward for the merge!

File: CMakeLists.txt
Comment (0x002A) [reply CMakeLists.txt:1]: Sounds totally reasonable to me. It was just the version I had at hand and I just wanted to know why downgrading the requirement makes sense. :)

**Review by 0x002A — APPROVED** [review]
Has Thomas already provided his feedback? For me its fine.

**Comment by TGatter** [comment]
You may go ahead an merge. I approve of the pipeline changes.

**Comment by 0x002A** [comment]
Your wish is my command.

**Pull request merged**
